frontend/ui/elements/screen_notification_menu_table.lua

```diff
@@ -0,0 +1,175 @@
+                end,
+                },
+            },
+        }
+    end
+    return ""
+end
+
+return {
+    text = _("Notification level"),
+    help_text = _([[KOReader may show notification popups at top of screen on various occasions.
+You can decide here what kind of notifications to show or hide.]]),
+    checked_func = function()
+        local value = G_reader_settings:readSetting("notification_sources_to_show_mask") or Notification.SOURCE_DEFAULT
+        return  value ~= 0
+    end,
+    sub_item_table = {
+        {
+        text = _("No popups"),
+        help_text = _("No popups are shown at all."),
+        checked_func = function()
+            return getMask() == Notification.SOURCE_NONE
+        end,
+        callback = function()
+            setMask(Notification.SOURCE_NONE)
+        end,
+        separator = true,
+        },
+        {
+        text = _("Only some bottom menu popus"),
+        help_text = _("Just show popups from the bottom menu, which don't have visual feedback."),
+        checked_func = function()
+            return band(getMask(), Notification.SOURCE_BOTTOM_MENU) == band(Notification.SOURCE_SOME, Notification.SOURCE_BOTTOM_MENU)
+        end,
+        callback = function()
+            if getMask() == Notification.SOURCE_ALL then
+                setMask(Notification.SOURCE_NONE)
+            end
+            setMask(bor(
+                    band(Notification.SOURCE_SOME, Notification.SOURCE_BOTTOM_MENU),
+                    band(getMask(), Notification.SOURCE_DISPATCHER)))
+        end,
+        },
+        {
+        text = _("A lot of bottom menu popus"),
+        help_text = _("Show popups from the bottom menu, which don't have visual feedback and popups from the three dots menus."),
+        checked_func = function()
+            return band(getMask(), Notification.SOURCE_BOTTOM_MENU) == band(Notification.SOURCE_DEFAULT, Notification.SOURCE_BOTTOM_MENU)
+        end,
+        callback = function()
+            if getMask() == Notification.SOURCE_ALL then
+                setMask(Notification.SOURCE_NONE)
+            end
+            setMask(bor(
+                    band(Notification.SOURCE_DEFAULT, Notification.SOURCE_BOTTOM_MENU),
+                    band(getMask(), Notification.SOURCE_DISPATCHER)))
+        end,
+        },
+        {
+        text = _("Gestures and profiles"),
+        help_text = _("Show popups on gestures and profiles."),
+        checked_func = function()
+            return band(getMask(), Notification.SOURCE_DISPATCHER) ~= 0 and getMask() ~= Notification.SOURCE_ALL
+        end,
+        callback = function()
+            if getMask() == Notification.SOURCE_ALL then
+                setMask(Notification.SOURCE_NONE)
+            end
+
+            setMask(bor(
+                    Notification.SOURCE_DISPATCHER,
+                    band(getMask(), Notification.SOURCE_BOTTOM_MENU)))
+        end,
+        separator = true,
+        },
```

Comment:
I'd clarify if that means *only* Gestures & Profiles, or if that's *in addition* to the previous level of verbosity.

(Unless the menu is *not* a mutually-exclusive set of choices?)

Reply:
The menu entries are designed so.
If you check "no popup" -> only this is selected
if you check "only some" -> "no popup" and "a lot of" are deselected 
if you check "a lot of" -> "no popup" and  "only some" are deselected
if you check "gestures" -> "no popup" is deselected

I think this is intuitive enough.